core/src/main/scala/org/apache/spark/api/python/PythonRunner.scala

```diff
@@ -57,9 +57,14 @@ private[spark] abstract class BasePythonRunner[IN, OUT](
 
   require(funcs.length == argOffsets.length, "argOffsets should have the same length as funcs")
 
+  private val env = SparkEnv.get
+  private val conf = env.conf
+
   // All the Python functions should have the same exec, version and envvars.
   protected val envVars = funcs.head.funcs.head.envVars
-  protected val pythonExec = funcs.head.funcs.head.pythonExec
+  protected val pythonExec = conf.getOption("spark.executorEnv.PYSPARK_DRIVER_PYTHON")
```

Comment:
Shouldn't this be handled in `deploy/PythonRunner`?

Reply:
Yeah, `PYSPARK_DRIVER_PYTHON` should be handled in `deploy/PythonRunner`.
This `PythonRunner` is for executors.